server/index.ts

```diff
@@ -1,122 +1,115 @@
     SocketEvent.JOIN_ROOM,
-    async (data: { roomCode: string; playerName: string }) => {
+    async ({
+      roomCode,
+      playerName,
+    }: {
+      roomCode: string;
+      playerName: string;
+    }) => {
       try {
         const { room, player } = await roomController.joinRoom(
-          data.roomCode,
+          roomCode,
           socket.id,
-          data.playerName
+          playerName
         );
         socket.join(room.roomCode);
         socket.broadcast.to(room.roomCode).emit(SocketEvent.NEW_PLAYER, player);
         socket.emit(SocketEvent.JOINED_SUCCESSFULLY, room);
       } catch (err) {
-        socket.emit("error", err.toString());
+        socket.emit('error', err.toString());
       }
     }
   );
 
   socket.on(SocketEvent.START_ROOM, async () => {
     try {
-      const room = await roomController.startGame(socket.id);
-      const { board, players } = room;
-      io.to(room.roomCode).emit(SocketEvent.ROOM_STARTED, { board, players });
+      const room = await roomController.startRoom(socket.id);
+      io.to(room.roomCode).emit(SocketEvent.ROOM_STARTED, {
+        board: room.board,
+        players: room.players,
+      });
     } catch (err) {
-      socket.emit("error", err.toString());
+      socket.emit('error', err.toString());
     }
   });
 
-  socket.on(SocketEvent.PLAY, async (data: { cards: Array<CardClass> }) => {
+  socket.on(SocketEvent.PLAY, async ({ cards }: { cards: Card[] }) => {
     try {
-      const { room, updated, player } = await roomController.playMove(
-        socket.id,
-        data.cards
-      );
-      io.to(room.roomCode).emit(SocketEvent.MOVE_PLAYED, {
-        room,
+      const {
+        name,
+        cards: playedCards,
         updated,
-        player,
+        board,
+        roomCode,
+      } = await roomController.playMove(socket.id, cards);
+      io.to(roomCode).emit(SocketEvent.MOVE_PLAYED, {
+        name,
+        cards: playedCards,
+        updated,
+        board,
       });
-    } catch (err) {}
+    } catch (err) {
+      socket.emit('eror', err.toString());
+    }
   });
 
-  socket.on("disconnect", async () => {
+  socket.on('disconnect', async () => {
     try {
-      const affectedRooms = await roomController.removePlayer(socket.id);
-      console.log(`${socket.id} disconnected.`);
-      if (!affectedRooms) {
-        console.error(
-          `No rooms will be notified of socket ${socket.id}'s disconnection`
-        );
+      const affectedRoom = await roomController.leaveRoom(socket.id);
+      if (!affectedRoom) {
         return;
       }
-      socket.leaveAll();
-      affectedRooms.forEach((affectedRoom) =>
-        io
-          .to(affectedRoom.roomCode)
-          .emit(SocketEvent.PLAYER_DISCONNECTED, socket.id)
+      io.to(affectedRoom.roomCode).emit(
+        SocketEvent.PLAYER_DISCONNECTED,
+        socket.id
       );
     } catch (err) {
-      socket.emit("error", err.toString());
+      socket.emit('err', err.toString());
```

Comment:
Also "error". Note for future: should we add what the error was in response to? Does not need to be done now.

Reply:
Good catch!